instrumentation.go

```diff
@@ -65,64 +62,62 @@ func NewInstrumentation(opts ...InstrumentationOption) (*Instrumentation, error)
 	}
 
 	c := newInstConfig(opts)
-	if err := c.validate(); err != nil {
-		return nil, err
-	}
-
-	pa := process.NewAnalyzer()
-	pid, err := pa.DiscoverProcessID(c.target)
-	if err != nil {
-		return nil, err
-	}
-
-	ctrl, err := opentelemetry.NewController(Version(), c.serviceName)
-	if err != nil {
-		return nil, err
-	}
 
-	mngr, err := instrumentors.NewManager(ctrl)
+	ctx := contextWithSigterm(context.Background())
+	log.Logger.V(0).Info("Establishing connection to OTLP receiver ...")
+	otlpTraceClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithDialOption(grpc.WithUserAgent(autoinstUserAgent)),
+	)
+	traceExporter, err := otlptrace.New(ctx, otlpTraceClient)
```

Comment:
Should this be passed as an option instead?

It seems like this could continue to be setup in the controller. It could be changed in a targeted follow up PR instead.

I think changing the exporter setup in this PR is going to bloat this PR. Especially since this doesn't seem like the appropriate way to expose the option to the user.

Reply:
We want to use same exporter for monitoring all the processes. If we do inside the controller, we will be creating on grpc connection for each process.